src/classes/Query.cls

```diff
@@ -1595,6 +1611,7 @@ public class Query {
     private Integer resultOffset = -1;
     private Map<String, Query> childQuerys = new Map<String, Query>();
     private Boolean addAllRows = false;
+    private Boolean isSystemContext = false;
 
     public class FieldSetting {
         public Boolean isAccessible = true;
```

Comment:
Hmm... I knew suggestion on this variable was coming.
How about creating `ParentFieldSetting` class initially with just one flag `isSystemContext`
 which can be extended in later enhancements if required.

using `isAccessible` again creates naming anomaly that I've been trying to address.
it's already doing what its named to do.

I am moving `isSystemContext` in `FieldSetting` & in new inner class `ParentFieldSetting` & using these two in respective methods.

Reply:
I am cool with using `ParentFieldSetting` at the moment, and we will fix it later.

But I think you still didn't understand my suggestion. Originally, `selectAllFields` will select only readable fields because of two lines:
`public Boolean isAccessible = true;` in here, and
`Boolean flag = config.isAccessible;` in the `addAllFields` method.

As you can see, all the other default values in `FieldSetting` are `false` but only `isAccessible` is `true`. So if we flip this to be `false`, same as the others, and also set `Boolean flag = true` in the `addAllFields` method, we remove the influence of `isAccessible`. As a result, all the fields will be selected. So, I am not trying to use `isAccessible`, instead I am trying to avoid using it.

Your solution, introducing another variable `isSystemContext` does work as expected. But in my opinion it creates variable redundancy. As we can using existing variables to represent the same information, I'd suggest not adding `isSystemContext` in `FieldSetting`.

Reply:
I understand your point, but following are the reasons I think `isSystemContext` works better.
1. Identical implementation for root object + parent object, so it's consistent.
2. Clearly states it's purpose. Having a initial `true` flag in `addAllFIelds` (& setting isAccessible = true from default false value only for `selectReadableFields` method) might confuse people, and they have to make assumptions/guess while making further changes, which may break functionalities or introduce unintended changes.

So, I will prefer keeping this variable for clarity in code, better readability & consistency. However, if you insist on not using it in Fieldsetting, please let me know. I will remove it.

Reply:
Yes, I agree with you for the first reason. The root object and the parent object should use the same implementation. Moreover, they should be using the same function, which requires some code refactor. I will work on that sometime later.

I see your confusions. Probably I should redesign the `FieldSetting` feature in version 2.0, too. I'll run a few more tests then I'll merge your PR.